### PR TITLE
Homework_8_Olha_Lesiuk

### PR DESCRIPTION
**Pull request by OlhaLesiuk**

### CODE REVIEW

File: Olha_Lesiuk/HW_8/task_1.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+It directly displays anything written to it to the Console. 
+"""
+
+
+def print_to_stdout(*args, **kwargs):
+    for arg in args:
+        print(*args, file=sys.stdout)
+    else:
+        for key, value in kwargs.items():
+            print("%s == %s" % (key, value))
```
Comment (bohdandovbsyh) [inline Olha_Lesiuk/HW_8/task_1.py]: you use print() function. It is not a solution. You should implement your own variant of this functionality

File: Olha_Lesiuk/HW_8/task_2.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+def my_own_filter(func, *iterable):
```
Comment (bohdandovbsyh) [inline Olha_Lesiuk/HW_8/task_2.py]: please add anotations

File: Olha_Lesiuk/HW_8/task_2.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    """
+    Implementing my own filter function.
+    """
+
+    data = []
+    if func is None:
+        for i in iterable:
```
Comment (bohdandovbsyh) [inline Olha_Lesiuk/HW_8/task_2.py]: naming!!!. what is i?

File: Olha_Lesiuk/HW_8/task_2.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+                    if is_even(x):
+                        data.append(x)
+    return data
+
+
+def is_even(x):
+    if x % 2 == 0:
+        return True
+    else:
+        return False
```
Comment (bohdandovbsyh) [inline Olha_Lesiuk/HW_8/task_2.py]: How does it work?

image

it did not return me what I want

File: Olha_Lesiuk/HW_8/task_4.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
+def maximum(x: list, amount_of_result=1):
+    """
+            Implementing my own max function.
+            """
+
+    max_number = x[0]
+    second_max_number = x[0]
+    if amount_of_result >= 2 or amount_of_result < 1:
+        raise Exception("The function should return 2 max values from the list")
```
Comment (bohdandovbsyh) [inline Olha_Lesiuk/HW_8/task_4.py]: why are only 2 numbers allowed? 2 in the task was as example not limitation.

File: Olha_Lesiuk/HW_8/task_4.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    elif amount_of_result == 2:
+        for i in x:
```
Comment (bohdandovbsyh) [inline Olha_Lesiuk/HW_8/task_4.py]: what is i and x?

File: Olha_Lesiuk/HW_8/task_4.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
+def maximum(x: list, amount_of_result=1):
```
Comment (bohdandovbsyh) [inline Olha_Lesiuk/HW_8/task_4.py]: naming. what is x?

File: Olha_Lesiuk/HW_8/task_4.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+def minimum(x: list, amount_of_result=1):
+    """
+            Implementing my own min function.
+            """
+
+    min_number = x[0]
+    second_min_number = x[0]
+    if amount_of_result >= 2 or amount_of_result < 1:
+        raise Exception("The function should return 2 max values from the list")
```
Comment (bohdandovbsyh) [inline Olha_Lesiuk/HW_8/task_4.py]: same here.

File: Olha_Lesiuk/HW_8/task_4.py [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+                min_number = i
+        for i in x:
+            if min_number < i < second_min_number:
+                second_min_number = i
+        return min_number, second_min_number
+    else:
+        for i in x:
+            if i < min_number:
+                min_number = i
+    return min_number
```
Comment (bohdandovbsyh) [inline Olha_Lesiuk/HW_8/task_4.py]: general comments. BAD naming. and please add return annotations

File: Olha_Lesiuk/HW_8/task_2.py (hunk 3 above)
Comment (OlhaLesiuk) [reply Olha_Lesiuk/HW_8/task_2.py]: i is index

File: Olha_Lesiuk/HW_8/task_4.py (hunk 6 above)
Comment (OlhaLesiuk) [reply Olha_Lesiuk/HW_8/task_4.py]: i is index and x is an array. I'm gonna fix it

File: Olha_Lesiuk/HW_8/task_1.py (hunk 1 above)
Comment (OlhaLesiuk) [reply Olha_Lesiuk/HW_8/task_1.py]: Fixed

File: Olha_Lesiuk/HW_8/task_2.py (hunk 3 above)
Comment (OlhaLesiuk) [reply Olha_Lesiuk/HW_8/task_2.py]: Fixed

File: Olha_Lesiuk/HW_8/task_4.py (hunk 5 above)
Comment (OlhaLesiuk) [reply Olha_Lesiuk/HW_8/task_4.py]: Changed logic

**Review by bohdandovbsyh — APPROVED** [review]

**Pull request merged**